containerd-shim-spin-config.md

```diff
@@ -0,0 +1,372 @@
+following categories of configuration:
+
+- Spin application execution configuration
+- Executor configuration
+
+The first category pertains to the execution of a single Spin application, while
+the later discusses how to configure the platform. 
+
+This SKIP focuses on how the `containerd-shim-spin` and `runwasi` will consume
+configuration. While this SKIP encourages the use of the SpinKube `SpinApp` and
+`SpinAppExecutor` CRDs to set configuration, it does not propose the structure
+of the changes to the CRDs to add configuration. That should be proposed in a
+future SKIP.
+
+### Spin Application Execution Configuration
+
+Spin application execution can be configured via container environment
+variables, which are exposed to the shim. The following is a subset of
+configuration values that may be supported:
+
+| Key                         | Spin CLI                                                    | Example Value           |
+|-----------------------------|-------------------------------------------------------------|-------------------------|
+| SPIN_HTTP_LISTEN_ADDR       | `spin up --listen`                                          | "0.0.0.0:3000"          |
+| OTEL_EXPORTER_OTLP_ENDPOINT | `OTEL_EXPORTER_OTLP_ENDPOINT=http://123.4.5.6:4318 spin up` | "http://123.4.5.6:4318" |
+| SPIN_LOG_DIR                | `spin up --log-dir /tmp/log`                                | "/tmp/log"              |
+| SPIN_RUNTIME_CONFIG_FILE    | `spin up --runtime-config-file /var/config.toml`            | "/var/config.toml"      |
+| AWS_DEFAULT_REGION          | `AWS_DEFAULT_REGION="us-west-2" spin up`                    | "us-west-2"             |
+| AWS_ACCESS_KEY_ID           | `AWS_ACCESS_KEY_ID="ABC" spin up`                           | "ABC"                   |
+| AWS_SECRET_ACCESS_KEY       | `AWS_SECRET_ACCESS_KEY="123" spin up`                       | "123"                   |
+| AWS_SESSION_TOKEN           | `AWS_SESSION_TOKEN="token" spin up`                         | "token"                 |
+
+These values will be configured as environment variables in the Container spec.
+For example, the following sets the listen address for the HTTP trigger:
+
+```sh
+spec
+  containers:
+  - name: myspinapp
+    image: "ghcr.io/myuser:v1"
+    env:
+    - name: SPIN_HTTP_LISTEN_ADDR
+      value: "0.0.0.0:3000"
+    ...
+```
+
+The `containerd-shim-spin` instance will look for execution environment
+variables and configure execution accordingly.
+
+At first, this feels strange, since it differs from user's previous
+understanding of Linux container environment variables. Deploying a Spin app
+shouldn't be compared to deploying a Linux container. A Linux container has one
+virtualized environment and listener. In a Linux container context, environment
+variables are set in the app execution and affect the business logic. On the
+other hand, a Spin app can have N many triggers and components, each of which
```

Comment:
You sort of allude to it but might be nice to be explicit that the spin has a host runtime (which is being called `Spin Application Execution` environment) and the components that it runs.   We don't want to share everything from the host to all components. Maybe the docs can do make this clearer too.